src/main/java/uk/gov/companieshouse/service/dissolution/DissolutionPatcher.java

```diff
@@ -23,28 +24,44 @@
 
     @Autowired
     public DissolutionPatcher(
-        DissolutionRepository repository,
-        DissolutionResponseMapper responseMapper,
-        DirectorApprovalMapper approvalMapper) {
+            DissolutionRepository repository,
+            DissolutionResponseMapper responseMapper,
+            DirectorApprovalMapper approvalMapper) {
         this.repository = repository;
         this.responseMapper = responseMapper;
         this.approvalMapper = approvalMapper;
     }
 
-    public DissolutionPatchResponse patch(String companyNumber, String userId, String ip, String email) {
+    public DissolutionPatchResponse addDirectorApproval(String companyNumber, String userId, String ip, String email) {
         final Dissolution dissolution = this.repository.findByCompanyNumber(companyNumber).get();
 
         this.addDirectorApproval(userId, ip, email, dissolution);
 
         if (!this.hasDirectorsLeftToApprove(dissolution)) {
-            this.setDissolutionToPendingPayment(dissolution);
+            dissolution
+                    .getData()
+                    .getApplication()
+                    .setStatus(ApplicationStatus.PENDING_PAYMENT);
         }
 
         this.repository.save(dissolution);
 
         return this.responseMapper.mapToDissolutionPatchResponse(companyNumber);
     }
 
+    public void updatePaymentStatus(String paymentReference, Timestamp paidAt, String companyNumber) {
```

Comment:
You are updating more than the payment status, maybe something like `addPaymentInformation` or `updatePaymentInformation` would be more appropriate?